### PR TITLE
New FindMKL module

### PR DESCRIPTION
**Pull request by teonnik**
The module uses targets instead of components and exposes many more MKL library combinations. It also searches more directories and is more portable.

For more information see the discussion here: https://github.com/eth-cscs/cmake-recipes/pull/8

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -51,7 +51,7 @@ if (NOT ${COSMA_BLAS} STREQUAL "CUDA" AND NOT ${COSMA_BLAS} STREQUAL "ROCM" AND
   endif()
 endif()
 
-project(cosma VERSION 2.3.0 LANGUAGES CXX)
+project(cosma VERSION 2.3.0 LANGUAGES C CXX)
```
Comment (kabicm) [inline CMakeLists.txt:54]: Is `C` language necessary?

**Review by kabicm — APPROVED** [review]
Looks great, thanks a lot @teonnik.

File: CMakeLists.txt (hunk 1 above)
Comment (kabicm) [reply CMakeLists.txt:54]: @rasolca mentioned in https://github.com/eth-cscs/COSMA/issues/69#issuecomment-756290398 that with gcc, find mpi requires both C and Fortran to be enabled. Is this required even if we use only the `CXX` component when finding MPI?

Comment (teonnik) [reply CMakeLists.txt:54]: For me, it works with CXX only and C is not needed. However, @rasolca did mention that not having C caused issues finding MPI, so I am not sure how to proceed.

I'm hesitant to add Fortran as a language as it brekas clang builds (and I can build fine without it)...

Comment (rasolca) [reply CMakeLists.txt:54]: I tried again and worked without C and Fortran. I then realized that last time I had a typo in my -DMPI_CXX_COMPILER argument when I tested with GCC.
It looks like that the missing C/Fortran problem arise when cmake doesn't find the mpicxx compiler and fallback to a different method.

Comment (teonnik) [reply CMakeLists.txt:54]: Alright, then, I think it's reasonable to expect that mpicxx is found. I will remove the `C` language parameter.

**Pull request merged**